packages/caliper-core/test/worker/caliper-worker.js

```diff
@@ -161,5 +190,101 @@ describe('Caliper worker', () => {
             await worker.executeRound(mockTestMessage);
             validateCallsAndWarnings(4);
         });
+
+        [5, 10].forEach(numberOfTxs => {
+            it(`should run ${numberOfTxs} transactions and wait for completion when no errors occur`, async () => {
+                const worker = new CaliperWorker(mockConnector, 1, mockMessenger, 'uuid');
+                await worker.prepareTest(mockTestMessage);
+
+                mockTestMessage.getNumberOfTxs.returns(numberOfTxs);
+                mockTestMessage.getRoundDuration.returns(null);
+
+                mockWorkload.submitTransaction.resetHistory();
+                mockStats.getTotalSubmittedTx.resetHistory();
+                mockStats.getTotalFinishedTx.resetHistory();
+                mockStats.getCumulativeTxStatistics.resetHistory();
+
+                let submittedTx = 0;
+                let finishedTx = 0;
+
+                // Stub the methods
+                mockStats.getTotalSubmittedTx.callsFake(() => submittedTx);
+                mockStats.getTotalFinishedTx.callsFake(() => finishedTx);
+                mockStats.getCumulativeTxStatistics.returns({});
+
+                worker.internalTxObserver.getCurrentStatistics = () => mockStats;
+
+                mockWorkload.submitTransaction.callsFake(async () => {
+                    submittedTx += 1;
+                    finishedTx += 1;
+                    return Promise.resolve();
+                });
+
+                await worker.executeRound(mockTestMessage);
+
+                sinon.assert.callCount(mockWorkload.submitTransaction, numberOfTxs);
+                sinon.assert.calledOnce(deactivateMethod);
+                sinon.assert.calledOnce(mockRate.end);
+                sinon.assert.calledOnce(mockWorkload.cleanupWorkloadModule);
+                sinon.assert.called(mockConnector.releaseContext);
+            });
+        });
+
+        it('should execute the round for a specified duration', async function() {
```

Comment:
tricky, the test doesn't prove that anything happened during this run and also doesn't prove that it took about 1 second. So you need to show that this test did do something for about the time period specified.

Reply:
what do you think about the update?

Reply:
yep, I think that's worth a try. Need to see how it behaves going forward to see if it ever fails the build being outside of the range.